<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <remap from="/plan/speed" to="/speed"/>
    <remap from="/plan/steering" to="/steering"/>

    <include file="$(find avc)/launch/mapper.launch"/>


    <arg name="n_path_segments" value="6"/>

    <node pkg="rr_common" type="planner" name="planner" output="screen">
        <param name="planner_type" value="annealing"/>

        <param name="collision_dist_front" value="0.62"/>
        <param name="collision_dist_back" value="0.17"/>
        <param name="collision_dist_side" value="0.25"/>
        <param name="obstacle_search_radius" value="3"/>

        <param name="wheel_base" value="0.485"/>
        <param name="lateral_accel" value="0.5"/>
        <param name="distance_increment" value="0.1"/>
        <param name="max_speed" value="0.5"/>
        <param name="max_steering" value="0.4"/>
        <param name="steering_speed" value="0.3"/>

        <param name="n_path_segments" value="$(arg n_path_segments)"/>
        <param name="segment_distances" value="$(eval ' '.join(['0.6'] * n_path_segments))"/>
        <param name="temperature_start" value="0.25"/>
        <param name="temperature_end" value="0.01"/>

        <param name="annealing_steps" value="500"/>
        <param name="k_dist" value="1.3"/>
        <param name="k_speed" value="1"/>
        <param name="k_similarity" value="0"/>
        <param name="k_final_pose" value="50"/>
        <param name="collision_penalty" value="1000"/>
        <param name="acceptance_scale" value="0.015"/>

        <param name="input_cloud_topic" value="/current_obstacles"/>

    </node>
</launch>
